src/event-db/src/types/snapshot.rs

```diff
@@ -0,0 +1,155 @@
+use crate::types::utils::serialize_systemtime_as_rfc3339;
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SnapshotVersion {
+    Latest,
+    Number(i32),
+    Name(String),
+}
+
+impl Serialize for SnapshotVersion {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Self::Latest => "latest".serialize(serializer),
+            Self::Number(number) => number.to_string().serialize(serializer),
+            Self::Name(name) => name.serialize(serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for SnapshotVersion {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct SnapshotVersionVisitor;
+        impl<'de> serde::de::Visitor<'de> for SnapshotVersionVisitor {
+            type Value = SnapshotVersion;
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str(
+                    r#"Expect one of the following options: "latest", "25", "Fund 10" etc."#,
+                )
+            }
+            fn visit_borrowed_str<E>(self, v: &'de str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                if v == "latest" {
+                    Ok(Self::Value::Latest)
+                } else {
+                    match v.parse::<i32>() {
+                        Ok(number) => Ok(Self::Value::Number(number)),
+                        _ => Ok(Self::Value::Name(v.to_string())),
+                    }
+                }
+            }
+        }
+        deserializer.deserialize_any(SnapshotVersionVisitor)
+    }
+}
+
+#[derive(Serialize, Clone, Default)]
+pub struct VoterInfo {
+    pub voting_power: i64,
+    pub voting_group: String,
+    pub delegations_power: i64,
+    pub delegations_count: i64,
+    pub voting_power_saturation: f64,
+}
+
+#[derive(Serialize, Clone)]
+pub struct Voter {
+    pub voter_info: VoterInfo,
+    #[serde(serialize_with = "serialize_systemtime_as_rfc3339")]
+    pub as_at: SystemTime,
+    #[serde(serialize_with = "serialize_systemtime_as_rfc3339")]
+    pub last_updated: SystemTime,
```

Comment:
I have a slight preference for `DateTime<Utc>` over `SystemTime`, because of the API it provides on the Rust side, but it's not a huge deal

Reply:
Fixed